ipatests/test_integration/test_acme.py

```diff
@@ -769,29 +824,213 @@ def test_pruning_negative_options(self):
 
         result = self.master.run_command(
             ['ipa-acme-manage', 'pruning',
-             '--cron="100 * * * *"'],
+             '--cron=100 * * * *'],
             raiseonerr=False
         )
         assert result.returncode == 1
         assert "100 not within the range 0-59" in result.stderr_text
 
         result = self.master.run_command(
             ['ipa-acme-manage', 'pruning',
-             '--cron="10 1-5 * * *"'],
+             '--cron=10 1-5 * * *'],
             raiseonerr=False
         )
         assert result.returncode == 1
         assert "1-5 ranges are not supported" in result.stderr_text
+
+    def test_prune_cert_manual(self, issue_and_expire_acme_cert):
+        """Test to prune expired certificate by manual run"""
+        if (tasks.get_pki_version(self.master)
+           < tasks.parse_version('11.3.0')):
+            raise pytest.skip("Certificate pruning is not available")
+
+        issue_and_expire_acme_cert(
+            self.master, self.clients[0], self.acme_server)
+
+        # check that the certificate issued for the client
+        result = self.master.run_command(['ipa', 'cert-find'])
+        assert 'CN=%s' % self.clients[0].hostname in result.stdout_text
+
+        # run prune command manually
+        self.master.run_command(['ipa-acme-manage', 'pruning', '--enable'])
+        self.master.run_command(['ipactl', 'restart'])
+        self.master.run_command(['ipa-acme-manage', 'pruning', '--run'])
+        time.sleep(100)
+
+        # check if client cert is removed
+        result = self.master.run_command(['ipa', 'cert-find'])
+        assert 'CN=%s' % self.clients[0].hostname not in result.stdout_text
+
+    def test_prune_cert_cron(self, issue_and_expire_acme_cert):
+        """Test to prune expired certificate by cron job"""
+        if (tasks.get_pki_version(self.master)
+           < tasks.parse_version('11.3.0')):
+            raise pytest.skip("Certificate pruning is not available")
+
+        issue_and_expire_acme_cert(
+            self.master, self.clients[0], self.acme_server)
+
+        # check that the certificate issued for the client
+        result = self.master.run_command(['ipa', 'cert-find'])
+        assert 'CN=%s' % self.clients[0].hostname in result.stdout_text
+
+        # enable pruning
+        self.master.run_command(['ipa-acme-manage', 'pruning', '--enable'])
+
+        # cron would be set to run the next minute
+        cron_minute = self.master.run_command(
+            [
+                "python3",
+                "-c",
+                (
+                    "from datetime import datetime; "
+                    "print(int(datetime.now().strftime('%M')) + 1)"
+                ),
+            ]
+        ).stdout_text.strip()
+        self.master.run_command(
+            ['ipa-acme-manage', 'pruning',
+             f'--cron={cron_minute} * * * *']
+        )
+        self.master.run_command(['ipactl', 'restart'])
+        time.sleep(150)
```

Comment:
Also fine for now, we'll need to collect actual data to determine how long to wait.